d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -282,6 +280,29 @@ private DualReadModeProvider.DualReadMode getDualReadMode(String d2ServiceName)
     return _dualReadStateManager.getServiceDualReadMode(d2ServiceName);
   }
 
+
+  /**
+   * Get the thread pool for load balancer tasks. If not set, a default thread pool will be used.
+   */
+  public ThreadPoolExecutor getLoadBalancerThreadPool() {
```

Comment:
After adding the threadpool to the constructor you can get rid of these two getters and setters

Reply:
Yeap. Updated